tests/unittest/_torch/auto_deploy/unit/singlegpu/transformations/library/test_moe_fusion.py

```
@@ -1074,3 +1074,253 @@ def test_nvfp4_moe_different_input_scales(
+
+# =============================================================================
+# FineGrainedFP8 MoE Tests
+# =============================================================================
+
+FP8_MAX = torch.finfo(torch.float8_e4m3fn).max
+
+
+class BlockSparseTop2MLPFineGrainedFP8(nn.Module):
+    """FineGrainedFP8 expert with per-block weight scales."""
+
+    def __init__(self, ffn_dim, hidden_dim, dtype=torch.bfloat16, device="cuda", block_size=None):
```

Ruff (0.15.1)

[warning] Unused method argument: `dtype`

(ARG002)